app/OsuApi.php

```diff
@@ -15,10 +20,26 @@ public function __construct(string $apiKey)
 
     private function call(string $resource, array $query)
     {
+        while (Cache::has(self::RATE_KEY)) {
+            usleep(1000000 * self::RATE_LIMIT + 1);
+        }
+
+        Cache::put(self::RATE_KEY, self::RATE_KEY, self::RATE_LIMIT / 60);
```

Comment:
depending on cache driver this might be a bad idea actually, these probably aren't designed for 0.1 second storage